vignettes/custom_epiworkflows.Rmd

```diff
@@ -0,0 +1,629 @@
+  add_role(time_value, new_role = "predictor") |>
+  # Use one-hot encoding on `geo_value_factor` and label each resulting column as a predictor
+  step_dummy(geo_value_factor) |>
+  # Create and lag `case_rate` growth rate
+  step_growth_rate(case_rate, role = "none", prefix = "gr_") |>
+  step_epi_lag(starts_with("gr_"), lag = c(0, 7, 14)) |>
+  step_epi_ahead(starts_with("gr_"), ahead = 7, role = "none") |>
+  # Divide growth rate into 3 bins, and label as outcome variable
+  # Note `recipes::step_cut()` has a bug that prevents us from using it here
+  step_mutate(
+    response = cut(
+      ahead_7_gr_7_rel_change_case_rate,
+      # Define bin thresholds.
+      # Divide by 7 to create weekly values.
+      breaks = c(-Inf, -0.2, 0.25, Inf) / 7,
+      labels = c("down", "flat", "up")
+    ),
+    role = "outcome"
+  ) |>
+  # Drop unused columns based on role assignments. This is not strictly
+  # necessary, as columns with roles unused in the model will be ignored anyway.
+  step_rm(has_role("none"), has_role("raw")) |>
+  step_epi_naomit()
+```
+
+This adds as predictors:
+
+- time value as a continuous variable (via `add_role()`)
+- `geo_value` as a set of indicator variables (via `step_dummy()` and the previous `as.factor()`)
+- growth rate of case rate, both at prediction time (no lag), and lagged by one and two weeks
+
+The outcome variable is created by composing several steps together. `step_epi_ahead()`
+creates a column with the growth rate one week into the future, and
+`step_mutate()` turns that column into a factor with 3 possible values,
+
+$$
+ Z_{\ell, t}=
+    \begin{cases}
+      \text{up}, & \text{if}\ Y^{\Delta}_{\ell, t} > 0.25 \\
+      \text{down}, & \text{if}\  Y^{\Delta}_{\ell, t} < -0.20\\
+      \text{flat}, & \text{otherwise}
+    \end{cases}
+$$
+
+where $Y^{\Delta}_{\ell, t}$ is the growth rate at location $\ell$ and time $t$.
+`up` means that the `case_rate` is has increased by at least 25%, while `down`
+means it has decreased by at least 20%.
+
+Note that in both `step_growth_rate()` and `step_epi_ahead()` we explicitly assign the role
+`none`. This is because those columns are used as intermediaries to create
+predictor and outcome columns.
+Afterwards, `step_rm()` drops the temporary columns, along with the original `role = "raw"` columns
+`death_rate` and `case_rate`. Both `geo_value_factor` and `time_value` are retained
+because their roles have been reassigned.
+
+
+To fit a classification model like this, we will need to use a `{parsnip}` model
+that has `mode = "classification"`.
+The simplest example of a `{parsnip}` `classification`-`mode` model is `multinomial_reg()`.
+The needed layers are more or less the same as the `linear_reg()` regression layers, with the addition that we need to remove some `NA` values:
+
+```{r, warning=FALSE}
+frost <- frosting() |>
+  layer_naomit(starts_with(".pred")) |>
+  layer_add_forecast_date() |>
+  layer_add_target_date() |>
+  layer_threshold()
```

Comment:
We don't need to threshold do we? Actually, this should probably be an error.
Let's add an issue.

Reply:
ha, yeah that doesn't make any sense. removed and filed https://github.com/cmu-delphi/epipredict/issues/465